app/screens/settings-screen/settings-screen.tsx

```diff
@@ -137,6 +140,27 @@ export const SettingsScreen: React.FC = () => {
       enabled: !isAuthed,
       greyed: isAuthed,
     },
+    {
+      category: LL.GaloyAddressScreen.yourAddress({ bankName: "BBW" }),
+      icon: "person",
+      id: "username",
+      subTitleDefaultValue: LL.SettingsScreen.tapUserName(),
+      subTitleText: lightningAddress,
+      action: () => {
+        Clipboard.setString(lightningAddress)
+        toastShow({
+          message: (translations) =>
+            translations.GaloyAddressScreen.copiedAddressToClipboard({
+              bankName,
+            }),
+          type: "success",
+          currentTranslation: LL,
+        })
+      },
+      enabled: isAuthed,
+      greyed: true,
```

Comment:
why is greyed always true? shouldn't it only be grey when it's already set?

Reply:
I know you're copying this around, but this doens't make much sense to me to grey it out if the button is a navigation action.

Reply:
oh also why was the action to set the username removed?

Reply:
I agree we should make the button grey. I'll fix that.
But we don't show this button if the user hasn't set the username, we only show it if they're logged in and when a username is present. They set the username from the next button `Receive payment using your bbw address`. The action to set username was not for this button in the first place.

Reply:
> But we don't show this button if the user hasn't set the username, we only show it if they're logged in and when a username is present. 

I think we should show it back. it's very intention to have a clear place to set your username. it's a key part of the onboarding, and currently it's otherwise "hidden" in a "merchants" feature

Reply:
> > But we don't show this button if the user hasn't set the username, we only show it if they're logged in and when a username is present.
> 
> I think we should show it back. it's very intentional to have a clear place to set your username. it's a key part of the onboarding, and currently it's otherwise "hidden" in a "merchants" feature

Yeah, that makes sense. I'll work on it.